### PR TITLE
add github action.

### PR DESCRIPTION
**Pull request by brendandburns**
cc @drubin

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '10', '8' ]
```
Comment (drubin) [inline .github/workflows/test.yml:14]: Any chance we can add ['12]

```suggestion
        node: [ '12', '10', '8' ]
```

As this is the latest LTS versions.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    name: Node ${{ matrix.node }} validation
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - run: npm install
```
Comment (drubin) [inline .github/workflows/test.yml:21]: Maybe we can add back 
https://github.com/kubernetes-client/javascript/blob/master/.travis.yml#L6-L9

```suggestion
      # Pre check to check that version matches package-lock.json
      # needs to happen before npm install
      - run: node version-check.js
      - run: npm install
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - run: npm test
+      - run: npm lint
```
Comment (drubin) [inline .github/workflows/test.yml:23]: Missing new line.

**Comment by drubin** [comment]
Can these be activated before merging? or I guess they can't

File: .github/workflows/test.yml (hunk 1 above)
Comment (brendandburns) [reply .github/workflows/test.yml]: done (removed '8')

File: .github/workflows/test.yml (hunk 2 above)
Comment (brendandburns) [reply .github/workflows/test.yml:22]: done.

File: .github/workflows/test.yml (hunk 3 above)
Comment (brendandburns) [reply .github/workflows/test.yml]: done.

**Comment by brendandburns** [comment]
Comments addressed. There is some tool that looks like it can run it locally (https://github.com/nektos/act) if I run into trouble validating this or need to make a bunch of changes, I will try that.

**Comment by brendandburns** [comment]
@drubin I think the `/lgtm` needs to be on it's own line :)

**Comment by drubin** [comment]
/lgtm

Sorry :)

**Pull request merged**